crates/nargo_cli/tests/test_data_ssa_refactor/regression/src/main.nr

```diff
@@ -0,0 +1,101 @@
+global NIBBLE_LENGTH: comptime Field = 16;
+
+fn compact_decode<N>(input: [u8; N], length: Field) -> ([u4; NIBBLE_LENGTH], Field)
+{
+    assert(2*input.len() as u64 <= NIBBLE_LENGTH as u64);
+    assert(length as u64 <= input.len() as u64);
+    
+    let mut nibble = [0 as u4; NIBBLE_LENGTH];
+
+    let first_nibble = (input[0] >> 4) as u4;
+    let parity = first_nibble as u1;
+    
+    if parity == 1
+    {
+	    nibble[0] = (input[0] & 0x0f) as u4;
+	    for i in 1..input.len()
+	    {
+ 	        if i as u64 < length as u64
+ 	        {
+	         	let x = input[i];
+ 	         	nibble[2*i - 1] = (x >> 4) as u4;
+	         	nibble[2*i] = (x & 0x0f) as u4;
+ 	        }
+	    }
+    }
+    else
+    {
+	    for i in 0..2
+	    {
+ 	        if (i as u64) < length as u64 - 1
+ 	        {
+		        let x = input[i + 1];
+ 		        nibble[2*i] = (x >> 4) as u4;
+		        nibble[2*i + 1] = (x & 0x0f) as u4;
+ 	        }
+	    }
+    }
+
+    let out = (nibble, 2*length + (parity as Field) - 2);
+
+    out
+}
+
+fn enc<N>(value: [u8; N], value_length: Field) -> ([u8; 32], Field)
+{
+    assert(value.len() as u8 >= value_length as u8);
+    let mut out_value = [0; 32];
+    if value_length == 0
+    {
+    	let out = (out_value, value_length);
+		out
+    }
+    else { if value_length as u8 < 31
+    {
+	out_value[0] = 0x80 + value_length as u8;
+	
+		for i in 1..value.len()
+		{
+			out_value[i] = value[i-1];
+		}
+		
+		let out = (out_value, value_length + 1);
+
+	out
+    }
+    else
+    {
+		let out = (out_value, 32);
+		out
+    }
+	}
+}
+
+fn main(x: [u8; 5], z: Field)
+{
+    //Issue 1144
+	let (nib, len) = compact_decode(x,z);
+	assert(len == 5);
+	assert([nib[0], nib[1], nib[2], nib[3], nib[4]] == [15, 1, 12, 11, 8]);
+
+
+}
+
+#[test]
+// Issue 1144
+fn test_1144()
```

Comment:
What is our stance on `#[test]` in these test cases? We don't run `nargo test` on them so these tests aren't actually tested automatically by the CI.
For this case specifically, it seems we're already testing for it on line 76 so I think we can remove this function.

Reply:
We probably don't want tests here as the default behavior is to not run tests, so they may get skipped